machine_learning/data_downloader.py

```diff
@@ -0,0 +1,160 @@
+import argparse
+import requests
+from urllib.parse import urlparse
+from web_monitoring.db import Client
+import pandas as pd
+import os
+import time
+from docopt import docopt
+import warnings
+
+def parse_url(url):
+    """
+    Parse Versionista change urls to get version ids
+    https://versionista.com/{site id}/{page id}/{to version id}:{from version id}
+
+    Parameters
+    ----------
+    url : string
+
+    Returns
+    -------
+    ids : dictionary
+        Site, page, and version ids of the from and two versions extracted from the url
+    """
+    try:
+        result = urlparse(url)
+        ids = [element for element in result.path.split('/') if element]
+        site_id = ids[0]
+        page_id = ids[1]
+        version_ids = ids[2].split(':')
+        to_version_id = version_ids[0]
+        from_version_id = version_ids[1]
+        ids = {'site_id': site_id, 'page_id': page_id,
+               'from_version_id': from_version_id, 'to_version_id': to_version_id}
+        return ids
+    except:
+        warnings.warn('Url "' + url + '" is not in the required format and no data will be downloaded for this row', Warning)
+
+def get_storage_uris(url):
+    """
+    Get uris of the versions stored in our db
+    Uses get_version_by_versionista_id from web_monitoring.db
+
+    Parameters
+    ----------
+    url : string
+
+    Returns
+    -------
+    from_version_uri, to_version_uri : string
+        To and from version uris to access content in db
+    ids : dictionary
+        Site, page, and version ids of the from and two versions extracted from the url
+    """
+    ids = parse_url(url)
+    from_version_id = ids['from_version_id']
+    to_version_id = ids['to_version_id']
+    print(ids)
+    client = Client.from_env()
+    """
```

Comment:
This is technically valid, but it would be better style to make it comment than a string. The _docstring_ immediately below the function signature is granted special significance by the language, but any additional comments further down should be comments, not strings.